src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
As this is task-specific ideally we wouldn't have this operation on `CDataStream` itself, but as a function that manipulates a CDataStream.